### PR TITLE
✨ Add referrer input screen

### PR DESCRIPTION
**Pull request by williamchong007**
the referrer input screen is borrowed from register liker id input screen

### CODE REVIEW

File: app/services/api/api.types.ts [hunk 1]
```diff
@@ -124,3 +124,15 @@ export type StatisticsTopSupportedCreatorsResult = {
     ids: string[]
   }
 } | GeneralApiProblem
+
+export type AppMetaResult = {
```
Comment (nwingt) [inline app/services/api/api.types.ts:128]: Please rename type to `UserAppMetaResult`

File: app/services/api/likeco-api.ts [hunk 2]
```diff
@@ -284,4 +284,33 @@ export class LikeCoAPI {
       return { kind: "bad-data" }
     }
   }
+
+  async fetchAppMeta(): Promise<Types.AppMetaResult> {
```
Comment (nwingt) [inline app/services/api/likeco-api.ts:288]: Please rename to `fetchUserAppMeta`

File: app/services/api/likeco-api.ts [hunk 3]
```diff
@@ -284,4 +284,33 @@ export class LikeCoAPI {
...
+    }
+
+    try {
+      return { kind: "ok", data: response.data }
+    } catch {
+      return { kind: "bad-data" }
+    }
+  }
+
+  async addAppReferrer(likerID: string): Promise<Types.GeneralResult> {
```
Comment (nwingt) [inline app/services/api/likeco-api.ts:303]: Please rename to `addUserAppReferrer`

File: app/models/app-meta/app-meta.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    isEmailVerified: types.optional(types.boolean, false),
+    firstOpenTs: types.optional(types.number, 0),
+    hasAndroid: types.optional(types.boolean, false),
+    hasIOS: types.optional(types.boolean, false),
+  })
+
+type AppMetaType = Instance<typeof AppMetaModel>
+export interface AppMeta extends AppMetaType {}
+type AppMetaSnapshotType = SnapshotOut<typeof AppMetaModel>
+export interface AppMetaSnapshot extends AppMetaSnapshotType {}
```
Comment (nwingt) [inline app/models/app-meta/app-meta.ts:19]: Please rename `AppMeta` to `UserAppMeta`

**Comment by williamchong007** [comment]
@nwingt rebased

**Comment by nwingt** [comment]
@williamchong007 Please review PR williamchong007/likecoin-app#1

**Pull request merged**